core/libraries/messages/EE_Message_To_Generate_From_Queue.php

```diff
@@ -32,6 +32,7 @@ public function __construct($messenger_name, $message_type_name, EE_Messages_Que
         if ($this->valid()) {
             $this->_message->set_content($this->_get_content());
             $this->_message->set_subject($this->_get_subject($custom_subject));
+            $this->_message->set_GRP_ID($this->_get_GRP_ID());
```

Comment:
this is technically fine but it _looks_ like you are passing the value returned from a getter to a setter, which would normally do absolutely nothing except eat up CPU cycles.

ie: getters and setters are _usually_named like:

 ```php
public function getProperty() {
	return $this->property;
}

public function setProperty( $value ) {
	$this->property = $value;
}
```

and your line of code here looks like:

```php
$obj->setProperty( $obj->getProperty() )
```

which would set the value of `property` using its current value (ie: no change).

Your `_get_GRP_ID()` function isn't a typical getter for the `GRP_ID` property, so it might be better to give it a name that clarifies that reality, such as:

```php
protected function getGroupIdFromMessageRepo()
```